### ISSUE TITLE
[Feature]: Beam search: top_p, min_p and logit processors

### ISSUE THREAD

**Issue by denadai2** [vllm-project/vllm#10754]
### 🚀 The feature, motivation and pitch

Dear vllm community, we recently deprecated beam search from the core library, in favour of a new method called `beam_search`. However, this new method is far less powerful than before and it restrict the possibilities of applying beam search to many use cases. For example, by controlling the generation (top_p etc) or doing constrained beam search (e.g. https://huggingface.co/blog/constrained-beam-search).

We, at Spotify, use 0.6.1 for this reason. I am sure that many more are doing the same. However, we would like to move to pytorch 2.5 to fully use our h100s, FSDP2 etc. Moreover, we would like to stay up-to-date with vllm

Could we consider moving these parameters in the new method as well? Thaaaaank you!

ref https://github.com/vllm-project/vllm/issues/6226

### Alternatives

Huggingface

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by mgoin** [comment]
Hey @denadai2 thanks for reporting. I think this is a resource priority problem since in theory you could pipe through anything to the internal SamplingParams created in the `beam_search` methods 
https://github.com/vllm-project/vllm/blob/4c05edb33ae4ae279421ddf981816d070e8ec37a/vllm/engine/protocol.py#L97-L101

Could you please narrow the list of potential parameters so someone can prioritize the meaningful ones? Contribution is welcome as well!

**Comment by youkaichao** [comment]
can you join the slack https://slack.vllm.ai for collaboration? 

I want to re-implement beam search in the flavor of parallel sampling, but do not have bandwidth.

see https://github.com/vllm-project/vllm/blob/4c05edb33ae4ae279421ddf981816d070e8ec37a/vllm/sequence.py#L1357

**Comment by njhill** [comment]
Yes IMO we should deprecate/remove the new beam_search method. We can make the original API work with the "external" implementation. No need for separate BeamSearchParams,  most SamplingParams can be used as-is, just need to adjust those that are themselves used by the beam search logic (like num logprobs) ... https://github.com/vllm-project/vllm/pull/9427#issuecomment-2430658034

**Comment by denadai2** [comment]
> can you join the slack https://slack.vllm.ai for collaboration?
> 
> I want to re-implement beam search in the flavor of parallel sampling, but do not have bandwidth.
> 
> see
> 
> https://github.com/vllm-project/vllm/blob/4c05edb33ae4ae279421ddf981816d070e8ec37a/vllm/sequence.py#L1357

Done! Is there a way to have a middle ground in which we do not have do do a complete refactor?

**Comment by stanislaushimovolos** [comment]
Hi!
@denadai2, @youkaichao
Are there any updates on this topic? I am also ready to help if needed!

**Comment by honne23** [comment]
Leaving a comment to follow progress on this feature
